src/js/shield_canvas_draw.js

```diff
@@ -317,18 +318,27 @@ export function diamond(fill, outline, ref, radius, outlineWidth, rectWidth) {
   return ctx;
 }
 
-export function homePlate(
+export function pentagon(
+  pointUp,
   offset,
+  angle,
   fill,
   outline,
   ref,
-  radius,
+  radius1,
+  radius2,
   outlineWidth,
   rectWidth
 ) {
+  let angleSign = pointUp ? -1 : 1;
+  let sine = Math.sin(angle);
+  let cosine = Math.cos(angle);
+  let tangent = Math.tan(angle);
+
   if (rectWidth == null) {
     var shieldWidth =
-      ShieldText.calculateTextWidth(ref, genericShieldFontSize) + 2 * PXR;
+      ShieldText.calculateTextWidth(ref, genericShieldFontSize) +
+      (2 + ((CS / PXR - offset) * tangent) / 2) * PXR;
```

Comment:
Same simplification as further up.  Given the same formula, wondering if there's a common aspect that can be refactored out.

Reply:
Fixed.